csrc/python_frontend/fusion_cache.cpp

```diff
@@ -258,15 +258,32 @@ FusionSchedules::FusionSchedules(int64_t fusion_id)
       last_user_def_scheduled_ir(nullptr),
       last_user_def_executor(nullptr),
       scheds_lock(),
-      fusion_id_{fusion_id} {
-  auto_gen_schedules = std::make_unique<FusionExecutorCache>(
-      std::make_unique<Fusion>(), fusion_id);
+      fusion_id_(fusion_id) {
+  presched_fusion_ = std::make_unique<Fusion>();
 }
 
 Fusion* FusionSchedules::preschedFusion() {
-  auto fusion = auto_gen_schedules->fusion();
-  NVF_CHECK(fusion != nullptr, "Prescheduled Fusion is unexpectedly null!");
-  return fusion;
+  if (presched_fusion_ != nullptr) {
+    return presched_fusion_.get();
+  }
+
+  // Ideally, we shouldn't have to access FusionExecutorCache::fusion() so
+  // FusionExecutorCache has the flexibility to modify it in place or even
+  // delete it. Currently, this is only needed for cloning an
+  // nvfuser.FusionDefinition. See exec_nvfuser's is_clonable parameter. After
+  // FusionDefinition.__exit__, FusionSchedules.presched_fusion_ is moved to
+  // FusionExecutorCache and therefore becomes null.
+  if (auto_gen_schedules != nullptr) {
+    return auto_gen_schedules->fusion();
+  }
+
+  NVF_THROW("Prescheduled Fusion is unexpectedly null!");
+}
+
+void FusionSchedules::createExecutorCache() {
+  auto_gen_schedules = std::make_unique<FusionExecutorCache>(
```

Comment:
Why not just clone `presched_fusion_` so `FusionExecutorCache` can always have a copy it can modify?

Reply:
Yes, a copy would increase latency, so we can skip it.